### ISSUE TITLE
Che server does not start

### ISSUE THREAD

**Issue by anujbhatia** [eclipse-che/che#217]
Downloaded che latest, unip and trying to launch using che.bat. It seems that startup script che.sh fails on the JDK version check (even though I have the required 1.8 version installed).

```
C:\Users\anbhatia\Documents\eclipse-che-latest\eclipse-che-4.0.0-beta-12\bin>che.bat

#############################################################
On Windows, Che projects can only reside in %userprofile% due
to limitations of Docker. On this computer, %userprofile% is
C:\Usersnbhatia
#############################################################

Docker machine named default already exists...
Setting environment variables for machine default...
Docker is configured to use vbox docker-machine named default with IP 192.168.99.100...

############## HOW TO CONNECT YOUR CHE CLIENT ###############
After Che server has booted, you can connect your clients by:
1. Open browser to http://localhost:8080, or:
2. Open native chromium app.
#############################################################

C:\Users\anbhatia\Documents\eclipse-che-latest\eclipse-che-4.0.0-beta-12\bin>
```

My environment - Win 7, JDK 1.8, Git 2.7.0:

```
C:\Users\anbhatia\Documents\eclipse-che-latest\eclipse-che-4.0.0-beta-12\bin>bash --version
GNU bash, version 4.3.42(5)-release (x86_64-pc-msys)
Copyright (C) 2013 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>

This is free software; you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

C:\Users\anbhatia\Documents\eclipse-che-latest\eclipse-che-4.0.0-beta-12\bin>ver

Microsoft Windows [Version 6.1.7601]
C:\Users\anbhatia\Documents\eclipse-che-latest\eclipse-che-4.0.0-beta-12\bin>"%JAVA_HOME%"\bin\java -version
java version "1.8.0_72"
Java(TM) SE Runtime Environment (build 1.8.0_72-b15)
Java HotSpot(TM) 64-Bit Server VM (build 25.72-b15, mixed mode)

C:\Users\anbhatia\Documents\eclipse-che-latest\eclipse-che-4.0.0-beta-12\bin>
```

**Comment by TylerJewell** [comment]
Hi.  Your setup looks good. Your bash version is a couple minor releases ahead of our test system, but cannot imagine that is the issue.

I do not see any errors on your startup. If the Java check failed, would have expected to see error output into the console. 

Have tested your adjustment and found no issues.

**Comment by ghost** [comment]
It looks like Che server has successfully started? Can you check of you have anything running on localhost:8080 after you execute che.bat? Sharing the entire log will also help, cause the chunk of log you have shared has no errors or clues that indicate possible issues.

**Comment by anujbhatia** [comment]
Yes I'm pretty sure the server does not start and I have nothing already running on port 8080. I'm not entirely sure why the helpful error message is not getting printed. I think it's something to do with the extra eval causing an error and the bash script terminating before the call to error_exit.

To test this I added extra debug echo statements to che.sh as below:

```
  echo JAVA_HOME = ${JAVA_HOME}
  echo Just before Version Check eval
  # Che requires Java version 1.8 or higher. 
  JAVA_VERSION=$(eval "${JAVA_HOME}"/bin/java -version 2>&1 | awk -F '"' '/version/ {print $2}')
  echo Just after version check eval
  if [[ "$JAVA_VERSION" < "1.8" ]]; then
      echo JAVA_VERSION = ${JAVA_VERSION}
      error_exit "Che requires Java version 1.8 or higher. We found a lower version."
      return
  fi
  echo aaaaa
```

Then I launched Git Bash and ran che.sh directly (instead of going through che.bat in a cmd prompt) using bash -x ./che.sh. The complete output is attached.
[che_sh_output.txt](https://github.com/codenvy/che/files/104415/che_sh_output.txt)

The last few lines of the output are:

```
+ echo Just before Version Check eval
Just before Version Check eval
++ eval '/c/Program Files/Java/jdk1.8.0_72/bin/java' -version
++ awk -F '"' '/version/ {print $2}'
+ JAVA_VERSION=

anbhatia@anbhatia-W7 MINGW64 ~/Documents/eclipse-che-latest/eclipse-che-4.0.0-beta-12/bin
```

The echo line just before the if statement isn't printed, execution stops at the line with the eval statement. So the error message isn't printed. I'm not sure exactly what's the fix to get the error message to print.

The script exits with exit code 127

**Comment by anujbhatia** [comment]
I think this has something to do with it:

```
anbhatia@anbhatia-W7 MINGW64 ~/Documents/eclipse-che-latest/eclipse-che-4.0.0-beta-12/bin
$ eval "${JAVA_HOME}"/bin/java -version 2>&1 | cat -
bash: c:Program: command not found

anbhatia@anbhatia-W7 MINGW64 ~/Documents/eclipse-che-latest/eclipse-che-4.0.0-beta-12/bin
$ eval '/c/Program Files/Java/jdk1.8.0_72/bin/java' -version
bash: /c/Program: No such file or directory
```

**Comment by TylerJewell** [comment]
1. If you remove the "eval" portion of the command, does it complete successfully?
2. If this persists, we can move the variables around to have the entire path to JAVA_HOME/bin/java be wrapped in quotes to account for windows boxes that have spaces in their path.  It looks like the eval statement on your version of bash is not properly handling the space in path.

**Comment by anujbhatia** [comment]
1. Yes removing the eval makes it work for me. The che server starts and stops as it should.
2. Spaces in the path is pretty common for windows. The Oracle JDK installer defaults to "C:\Program Files..." or "C:\Program FIles (x86)..."  unless the user selects a different location of course. As for eval - I'm not an expert but to me it looked like eval was doing what it was supposed to and it use in the shell script was a bug. But it's your call on what's the right way to fix it. It could be just something in my environment.

**Comment by garagatyi** [comment]
I agree that `eval` was used by mistake. I fixed it in #223

**Comment by TylerJewell** [comment]
LGTM.  I think the consistency applied to all variables was helpful, too.
